### PR TITLE
Rework Muon candidate selection in few tracker ALCARECO  

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

This PR is a partial response to https://github.com/cms-sw/cmssw/issues/46493.
In https://github.com/cms-sw/cmssw/issues/46493#issuecomment-2432672540 it was suggested to replace few of the `MuonSelector` (`ObjectSelectorBase<SingleElementCollectionSelector<edm::View<reco::Muon>, ...> `) instances with modules that express the structure of the selection in C++ code instead, as this would allow to reduce the memory footprint in Run 3 prompt reco jobs for `Muon*` PDs.
This is implemented in commits 8e3de6568ee2a4d2b6141d8463689274fd12e0a0 and 2d97abd56f34981d62e260bcc10725662379cd41 that introduce ad-hoc classes `AlignmentGoodIdMuonSelector` and `AlignmentRelCombIsoMuonSelector` for this purpose.
Additionally in https://github.com/cms-sw/cmssw/issues/46493#issuecomment-2432744958 it was suggested that the duplication of selection identical modules configuration has both memory and runtime cost. This is addressed at 8e49f10451877c56f6cf1def481603198b840914 by introducing a new common sequence `seqALCARECOTkAlRelCombIsoMuons` that ulimately takes in input a `reco::MuonCollection` (with label `muons`) and returns a skimmed `reco::MuonCollection` called `TkAlRelCombIsoMuonSelector` which is then used to feed the rest of the selection chain.

#### PR validation:

```
runTheMatrix.py -l 1001.0,1001.2,1001.3,1001.4,1002.4
```
runs. 

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * workflows = 1001.0,1001.2,1001.3,1001.4,1002.4

**Comment by mmusich** [comment]
type performance-improvements

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Alignment/CommonAlignmentProducer/python/TkAlMuonSelectors_cfi.py [hunk 1]
```diff
@@ -1,31 +1,37 @@
...
-)
+from Alignment.CommonAlignmentProducer.alignmentGoodIdMuonSelector_cfi import alignmentGoodIdMuonSelector
+TkAlGoodIdMuonSelector = alignmentGoodIdMuonSelector.clone(src = cms.InputTag('muons'),
+                                                           requireGlobal = cms.bool(True),
+                                                           requireTracker = cms.bool(True),
+                                                           minMatches = cms.int32(1),
+                                                           minMuonHits = cms.int32(0),
+                                                           maxEta = cms.double(2.5),
+                                                           maxChi2 = cms.double(20),
+                                                           filter = cms.bool(True))
```
Comment (Dr15Jones) [inline Alignment/CommonAlignmentProducer/python/TkAlMuonSelectors_cfi.py:11]: You should be able to make this even simpler by doing
```python
AlGoodIdMuonSelector = alignmentGoodIdMuonSelector.clone(src = 'muons',
                                                           requireGlobal = True,
                                                           requireTracker = True,
                                                           minMatches = 1,
                                                           minMuonHits = 0,
                                                           maxEta = 2.5,
                                                           maxChi2 =20,
                                                           filter = True)
```
Dropping the types should work for all the configurations you have made.

**Review by Dr15Jones — COMMENTED** [review]
Thanks for this!

File: Alignment/CommonAlignmentProducer/python/TkAlMuonSelectors_cfi.py (hunk 1 above)
Comment (mmusich) [reply Alignment/CommonAlignmentProducer/python/TkAlMuonSelectors_cfi.py]: Thanks for the suggestion, implemented in the last push.

**Comment by mmusich** [comment]
test parameters:
   * enable_tests = threading
   * workflows_threading = 1000.0, 1001.0,1001.2,1001.3,1001.4,1002.4

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by perrotta** [comment]
+alca

**Pull request merged**